Corona-Warn-App/src/main/java/de/rki/coronawarnapp/transaction/SubmitDiagnosisKeysTransaction.kt

```diff
@@ -54,39 +54,42 @@ object SubmitDiagnosisKeysTransaction : Transaction() {
         AppInjector.component.transSubmitDiagnosisInjection.transactionScope
     }
 
+    private val playbook: Playbook
+        get() = AppInjector.component.transSubmitDiagnosisInjection.playbook
+
+    private val appConfigProvider: AppConfigProvider
+        get() = AppInjector.component.transSubmitDiagnosisInjection.appConfigProvider
+
     /** initiates the transaction. This suspend function guarantees a successful transaction once completed. */
     suspend fun start(
         registrationToken: String,
         keys: List<TemporaryExposureKey>,
         symptoms: Symptoms
     ) = lockAndExecute(unique = true, scope = transactionScope) {
-        /****************************************************
-         * RETRIEVE TEMPORARY EXPOSURE KEY HISTORY
-         ****************************************************/
+
         val temporaryExposureKeyList = executeState(RETRIEVE_TEMPORARY_EXPOSURE_KEY_HISTORY) {
             ExposureKeyHistoryCalculations(
                 TransmissionRiskVectorDeterminator(),
                 DefaultKeyConverter()
             ).transformToKeyHistoryInExternalFormat(keys, symptoms)
         }
-        /****************************************************
-         * RETRIEVE TAN & SUBMIT KEYS
-         ****************************************************/
+
+        val visistedCountries = appConfigProvider.getAppConfig().supportedCountriesList
+
         executeState(RETRIEVE_TAN_AND_SUBMIT_KEYS) {
-            PlaybookImpl(WebRequestBuilder.getInstance()).submission(
-                registrationToken,
-                temporaryExposureKeyList
+            val submissionData = Playbook.SubmissionData(
+                registrationToken = registrationToken,
+                temporaryExposureKeys = temporaryExposureKeyList,
+                consentToFederation = true,
+                visistedCountries = visistedCountries
```

Comment:
Question: Is the fallback to `DE` in case `visistedCountries` is empty built into the App Config?

Reply:
It is not. Good point. Will fix.